architecture-decision-records/eval-hub/ODH-ADR-EH-0004-eval-sidecar-creds-injection.md

```
@@ -0,0 +1,223 @@
+  hf-token: aGZfSFJOd2lHV2huVk9XbHFRVE...
+```
+
+`hf-token` is not used for the model but rather for the gated eval datasets. 
+
+
+Our goal here is to not share the actual user's API_KEY to the eval container as today. In order to do this we propose the following flow:
+
+1. Nothing changes the evalhub REST API. The users still provides the same reference to the secret that containes the actual API_KEY
+2. EvalHub service generates the following secret:
+
+#### Secret 1 
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h3; Actual: h4

(MD001, heading-increment)